### DISCUSSION TITLE
Middleware only picking GET requests, but not POST requests

### DISCUSSION THREAD

**Discussion by ObaidAshiq** [vercel/next.js#62390] [Help] [answered]
### Summary

I am not sure but my fetch requests to another nodeJs/express server with method post are being not picked up by it 

    export function middleware(request) {
      console.log(request.method)
     }
  This file is located as -(Using pages router)
  src/middleware.js

**Comment by icyJoseph** [comment]
> my fetch requests to another nodeJs/express server 

You mean, that, from the client, you are sending a POST request to an external server? And that somehow the GET request, from the client to the same external server, triggers the middleware?

**Reply by ObaidAshiq** [reply]
What i mean , i have added middleware.js file
So i was expecting it to pick those requests also that i am making to other server in some components

**Reply by ObaidAshiq** [reply]
Suppose i have a component which uses fetch api  to make a request of method of post -It gets users from a different origin, say localhost:9002
My next app is hosted at say loclahost:3000,, my middleware is able to run on every GET method(like the icons), but i dont see it working to the POST request that i made from fetch api to the localhost:9002

**Reply by ObaidAshiq** [reply]
This is what my next.confug.js looks like

    /** @type {import('next').NextConfig} */
     const nextConfig = {
        reactStrictMode: true,
    }
    module.exports = nextConfig

**Reply by icyJoseph** [reply, marked as answer]
Hi,

I am have gotten a bit busy now, but I can say that middleware would only intercept request you send to your own origin... That's because the fetch request created on the client, is independent of your own. I am not sure how to put it other terms, but as far as that fetch request is concerned, your Next.js server (and its middleware) doesn't even exist.

You could try to rewrite the request though, so that you'd fire the request against your own domain, and rewrite it to the other service. And if I remember correctly, that'll trigger the middleware!

**Comment by Eunkyung-Son** [comment]
You can access middleware through server action.
